app/src/ui/stashing/confirm-discard-stash.tsx

```diff
@@ -0,0 +1,78 @@
+import React = require('react')
+import { Dialog, DialogContent, DialogFooter } from '../dialog'
+import { Repository } from '../../models/repository'
+import { Dispatcher } from '../dispatcher'
+import { ButtonGroup } from '../lib/button-group'
+import { Button } from '../lib/button'
+import { Row } from '../lib/row'
+import { IStashEntry } from '../../models/stash-entry'
+
+interface IDiscardStashProps {
+  readonly dispatcher: Dispatcher
+  readonly repository: Repository
+  readonly stash: IStashEntry
+  readonly onDismissed: () => void
+}
+
+interface IDiscardStashState {
+  readonly isDiscarding: boolean
+}
+/**
+ * Dialog to confirm dropping a stash
+ */
+export class DiscardStash extends React.Component<
```

Comment:
:art: we use the `Confirm` prefix in other places to make the intent clearer. Also, maybe make clear this is rendering a dialog?

```suggestion
export class ConfirmDiscardStashDialog extends React.Component<
```

PS This will break a bunch of code, only take it in if you really want to do that 😁

Reply:
0bb3b1f

Reply:
also f324e60